vcd/datasource_vcd_independent_disk_test.go

```diff
@@ -0,0 +1,114 @@
+// +build disk ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/terraform"
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+// Test independent disk data resource
+// Using a disk data source we reference a disk data source
+func TestAccVcdDataSourceIndependentDisk(t *testing.T) {
+	resourceName := "TestAccVcdDataSourceIndependentDisk_1"
+	datasourceName := "TestAccVcdDataSourceIndependentDisk_Data"
+	diskName := "TestAccVcdDataSourceIndependentDisk"
+
+	var params = StringMap{
+		"Vdc":                testConfig.VCD.Vdc,
+		"name":               diskName,
+		"description":        diskName + "description",
+		"size":               "5242880",
+		"busType":            "SCSI",
+		"busSubType":         "lsilogicsas",
+		"storageProfileName": "*",
+		"ResourceName":       resourceName,
+		"Tags":               "disk",
+		"dataSourceName":     datasourceName,
+	}
+
+	configText := templateFill(testAccCheckVcdDataSourceIndependentDisk, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testDiskResourcesDestroyed,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config:             configText,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDiskCreated("vcd_independent_disk."+resourceName, diskName),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "name", diskName),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "description", diskName+"description"),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "size_in_bytes", "5242880"),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "bus_type", "SCSI"),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "bus_sub_type", "lsilogicsas"),
+					resource.TestCheckResourceAttr("data.vcd_independent_disk."+datasourceName, "storage_profile", "*"),
+					resource.TestMatchOutput("owner_name", regexp.MustCompile(`^\w*$`)),
```

Comment:
This regular expression means "match a string of zero or more word characters". Thus it will match "abc", or "a_bc", but also the empty string "". However, it won't match "ab-c"

Reply:
updated to `non-empty`